### ISSUE TITLE
when adding of a provider account fails, I want a sane error message...

### ISSUE THREAD

**Issue by martinpovolny** [aeolusproject/conductor#139]
## What I want

when adding of a provider account fails, I want a sane error message telling me details of what went wrong and how to get from there (where to look at least)
## What I get

error message: "Cannot add the Provider Account." telling me really nothing
- Conductor (Rails) log, tells me nothing
- quick look into /var/log/messages!! to find out I forgot to disable selinux

DC has a bug filled (and maybe already closed) on sane logfile location and sane logfile content

(rant issue # 2)

**Comment by mfojtik** [comment]
Having DC log does not really help Conductor error message look better. What we can do here is actually parse the DC logs or save the exceptions (DC do return exceptions for all error responses) and then give user a link where he can get more details about the error.

Do you have any idea how this can be implemented for Conductor?

**Comment by razorinc** [comment]
I agree with @mifo here. We cannot change the deltacloud errors to fit conductor so seems that conductor should figure out a way to manage the errors.

**Comment by martinpovolny** [comment]
Well guys at least you can tell the user what component went wrong and where to look.

If there's no nice message or a known error code from a backend, you shall at least provide the user with a hint of what to do next.

e.g. "Failed to create a Provider Account. Details: Delta Cloud failed with error #65535. Please check Delta Cloud log in /var/log/deltacloud/fail.log"

**Comment by jcoufal** [comment]
@martinpovolny Martin, it's WE, not you :)

Anyway, Martin is right. There should be some functionality to recognize the problem in the background and translate the problem to **user-readable** and **user-understandable** way, so he knows what happened and how to continue to resolve the problem.

**Comment by mfojtik** [comment]
@jcoufal +1

Pointing user to location where he can find DC logs sounds like a good start :-)

**Comment by aathomas** [comment]
Seems to me there are two parts to this
- the most comprehensive account possible of what really went wrong should be accessible from deltacloud
- Conductor should access that information, in whatever way the deltacloud folks recommend, and then use it to inform the user via the UI..

Telling the user that they should read the logs is (a bit) better than nothing, but the really interesting question is, how much information does DC export, and how much more could it export?

**Comment by mfojtik** [comment]
@aathomas Deltacloud do report all errors/exceptions in XML response. So basically you get HTTP status error (50x) and you also get detailed error description with full backtrace back wrapped in XML response.

AFAIK conductor is not using this right now. There are some improvements in dc client code that make these errors report as a Ruby exceptions. So if Conductor can pass these exceptions to user then user get more detailed errors of what went wrong.

**Comment by aathomas** [comment]
@mifo Thanks. Sounds like that's exactly what we need to do.

**Comment by jcoufal** [comment]
@mifo Are they somehow not-very-skilled-user-understendable or more technical ones?

**Comment by mfojtik** [comment]
@jcoufal They are more technical. Exception messages usually contain what backend cloud returned as a error messages. If there is no message from the backend, then we translate the HTTP status code to human language, typically for the client errors (4xx), like (405 - Method not allowed for this resource.).

There is a JIRA ticket[1] that @jprovaznik opened a while ago where we're trying improve and unify the exceptions across all cloud providers for different 'common' errors, like "Instance could not be started, because of incompatible hardware profile" and so. 

[1] https://issues.apache.org/jira/browse/DTACLOUD-309

**Comment by jcoufal** [comment]
@mifo +1 for JIRA ticket - that's what we need across whole Conductor.

Example: Instance could not be started, because of incompatible hardware profile (+ at least link where to fix that, better to say also how).
